test/integration/addons_test.go

```diff
@@ -190,7 +190,74 @@ func testServicesList(t *testing.T) {
 		t.Fatalf(err.Error())
 	}
 }
+func testRegistry(t *testing.T) {
+	t.Parallel()
+	minikubeRunner := NewMinikubeRunner(t)
+	kubectlRunner := util.NewKubectlRunner(t)
+	minikubeRunner.RunCommand("addons enable registry", true)
+	t.Log("wait for registry to come up")
+
+	if err := util.WaitForDockerRegistryRunning(t); err != nil {
+		t.Fatalf("waiting for registry to be up: %v", err)
+	}
+
+	// Check access from outside the cluster on port 5000, validing connectivity via registry-proxy
+	checkExternalAccess := func() error {
+		t.Log("checking registry access from outside cluster")
+		_, out := minikubeRunner.RunDaemon("ip")
+		s, err := readLineWithTimeout(out, 180*time.Second)
 
+		if err != nil {
+			t.Fatalf("failed to read minikubeIP: %v", err)
+		}
+
+		registryEndpoint := "http://" + strings.TrimSpace(s) + ":5000"
+		u, err := url.Parse(registryEndpoint)
+
+		if err != nil {
+			t.Fatalf("failed to parse %q: %v", s, err)
+		}
+
+		resp, err := retryablehttp.Get(u.String())
+		if err != nil {
+			t.Fatalf("failed get: %v", err)
```

Comment:
the t.Fatalf will make the test fail, and will not give it a chace to be retried by util.Retry.
you could either return error or not do the util.Retry at all.